moped-editor/src/views/projects/projectView/SignalComponentAutocomplete.js

```diff
@@ -0,0 +1,137 @@
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+};
+
+/*
+/ Hook which updates the component editor state based on the selected signal
+*/
+const useSignalChangeEffect = (
+  signal,
+  setSelectedComponentSubtype,
+  setEditFeatureCollection
+) => {
+  useEffect(() => {
+    const signalSubtype = signal
+      ? signal.properties.signal_type.toLowerCase()
+      : "";
+    const featureCollection = signalToFeatureCollection(signal);
+    setSelectedComponentSubtype(signalSubtype);
+    setEditFeatureCollection(featureCollection);
+  }, [signal, setSelectedComponentSubtype, setEditFeatureCollection]);
+};
+
+/**
+ * Material Autocomplete wrapper that enables selecting a traffic/phb signal record from a
+ * Socrata dataset and setting it as a project component.
+ * * @param {Object} setSelectedComponentSubtype - sets the selectedComponentSubtype from parent state
+ * * @param {Object} setEditFeatureCollection - sets the editFeatureCollection from parent state
+ * * @param {Boolean} editFeatureCollection - the geoJSON generated for all the the features in this component
+ * * @param {Object} classes - MaterialUI style object
+ *  @return {JSX.Element}
+ */
+const SignalComponentAutocomplete = ({
+  classes,
+  setSelectedComponentSubtype,
+  setEditFeatureCollection,
+  editFeatureCollection,
+}) => {
+  const [signal, setSignal] = useState("");
+  const { features, loading, error } = useSocrataGeojson(SOCRATA_ENDPOINT);
+  const options = features ? [...features, ""] : [""];
+
+  useInitialComponentValue(editFeatureCollection, setSignal);
+
+  useSignalChangeEffect(
+    signal,
+    setSelectedComponentSubtype,
+    setEditFeatureCollection
+  );
+
+  if (loading) {
+    return <CircularProgress color="primary" size={20} />;
+  } else if (error) {
+    return (
+      <Alert severity="error">{`Unable to load signal list: ${error}`}</Alert>
+    );
+  }
+
+  return (
+    <Autocomplete
+      className={classes}
+      id="signal-id"
+      filterOptions={(options, { inputValue, getOptionLabel }) => {
+        // limits options to ensure fast rendering
+        const limit = 40;
+        // applies the default autcomplete matching behavior plus our limit filter
+        const filteredOptions = options.filter(option =>
+          getOptionLabel(option)
+            .toLowerCase()
+            .includes(inputValue.toLowerCase())
+        );
+        return filteredOptions.slice(0, limit);
+      }}
+      getOptionSelected={(option, value) => {
+        return value
+          ? option.properties?.signal_id === value.properties?.signal_id
+          : option === "";
+      }}
+      // this label formatting mirrors the Data Tracker formatting
+      getOptionLabel={option =>
+        option
+          ? `${option.properties.signal_id}: ${option.properties.location_name}`
+          : ""
+      }
+      onChange={(e, signal) => {
+        setSignal(signal);
+      }}
+      loading={loading}
+      options={options}
```

Comment:
Maybe options need to be filtered to have a length of at least one character?

Reply:
i over-engineered this because i didn't understand the normal MUI autocomplete behavior. will refactor.